src/libNode/FinalBlockProcessing.cpp

```diff
@@ -155,6 +155,8 @@ bool Node::LoadUnavailableMicroBlockHashes(const TxBlock& finalBlock,
           if (m_microblock->GetBlockHash() == info.m_microBlockHash) {
             if (m_microblock->GetHeader().GetTxRootHash() != TxnHash()) {
               if (info.m_txnRootHash != TxnHash()) {
+                // Update transaction processed
+                UpdateProcessedTransactions();
```

Comment:
This seems out of place in this PR.  Is it a fix for some issue?

Reply:
Yes it is. The purpose is to let the shard node update the processed txns only when they found their microblock in the finalblock, while the adoption now is update when the microblock consensus done, but it could happen even the consensus was done, but the microblock may still not be inside of the finalblock.